lib/repo-url.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { URL } = require('url')
+const hostedGitInfo = require('hosted-git-info')
+
+/**
+ * @param packageName A package name as listed in package.json's dependencies list
+ * @param packageJson Optional param to specify a object representation of a package.json file instead of loading from node_modules
+ * @returns A valid url to the root of the package's source or null if a url could not be determined
+ */
+function getRepoUrl(packageName, packageJson = undefined) {
```

Comment:
`packageJson = undefined` can be simplified to `packageJson`

Reply:
I took this approach to make it explicit that it was optional. Having a default value (even a redundant one) will result in the IDE tooltip for this function to be:

```
getRepoUrl(packageName: any, packageJson?: any): any
```

rather than

```
getRepoUrl(packageName: any, packageJson: any): any
```

If you'd like it to be removed and rely on the JSDoc to explain that it's optional, I'll make the change.

Reply:
Interesting! Well that's preferable until I get around to converting to Typescript :)